deprecation_notifier/DeprecationNotifier/DNAppDelegate.m

```diff
@@ -42,15 +42,26 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
   NSDictionary *systemVersionDictionary = [NSDictionary dictionaryWithContentsOfFile:
                                               @"/System/Library/CoreServices/SystemVersion.plist"];
   NSString *systemVersion = systemVersionDictionary[@"ProductVersion"];
-
+  
   NSArray *systemVersionArray = [systemVersion componentsSeparatedByString:@"."];
+  NSMutableArray *mutableSystemVersionArray = [NSMutableArray array];
+  [mutableSystemVersionArray addObjectsFromArray:systemVersionArray];
+  if (mutableSystemVersionArray.count == 2) {
+      [mutableSystemVersionArray addObject:@"0"];
+  }
```

Comment:
This should work well but you can avoid making a mutable copy of the array when it isn't necessary by doing:

```objc
NSArray *systemVersionArray = [systemVersion componentsSeparatedByString:@"."];
if (systemVersionArray.count == 2) {
  systemVersionArray = [systemVersionArray arrayByAddingObject:@"0"];
}
```

This can apply to both the arrays and then you can use systemVersionArray and expectedVersionArray in the conditions further down.